addons/controllers/csi/csiconfig_datavalues.go

```diff
@@ -0,0 +1,53 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+// DataValues is the data values type generated by the CSIConfig CR
+type DataValues struct {
+	VSphereCSI   *DataValuesVSphereCSI   `yaml:"vsphereCSI,omitempty"`
+	VSpherePVCSI *DataValuesVSpherePVCSI `yaml:"vspherePVCSI,omitempty"`
+}
+
+type DataValuesVSpherePVCSI struct {
+	ClusterName                      string `json:"clusterName"`
+	ClusterUID                       string `json:"clusterUID"`
+	Namespace                        string `json:"namespace"`
+	SupervisorMasterEndpointHostname string `json:"supervisorMasterEndpointHostname"`
+	SupervisorMasterPort             int32  `json:"supervisorMasterPort"`
+}
+
+type DataValuesVSphereCSI struct {
+	Namespace             string  `json:"namespace"`
+	ClusterName           string  `json:"clusterName"`
+	Server                string  `json:"server"`
+	Datacenter            string  `json:"datacenter"`
+	PublicNetwork         string  `json:"publicNetwork"`
+	Username              string  `json:"username"`
+	Password              string  `json:"password"`
+	Region                *string `json:"region,omitempty"`
```

Comment:
Non-blocking: Using pointer types will introduce more trouble when accessing them (nilness check). Using value type still work with `omitempty`, as for string, "" will be omitted when being marshalled.

Reply:
updated